## Titanic Data:

Table of contents:
1. Importing Libraries.
2. Importing Datasets
3. Perfroming Basic EDA(Peek into data)
4. Correlation between 'Sex' and 'Survived'.

#### 1.Importing Libraries

In [1]:
import pandas as pd
import numpy as np

#### 2.Importing Datasets

In [2]:
train = pd.read_csv('train.csv')
test = pd.read_csv('test.csv')

#### 3.Peek into data:

##### a.train.columns.values -- gives the total column names in the data
##### b.train.describe() -- gives the mean, standard deviation, min and max values for int type features.
##### c..train.isnull().sum() -- gives the total number of null rows in a column

In [3]:
train.columns.values

array(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'], dtype=object)

In [4]:
train.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [5]:
train.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [6]:
test.columns.values

array(['PassengerId', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch',
       'Ticket', 'Fare', 'Cabin', 'Embarked'], dtype=object)

In [7]:
test.isnull().sum()

PassengerId      0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

#### Finding corelation between Pclass,SibSp,Parch and Survived classes :

In [8]:
print(train[['Pclass', 'Survived']].groupby(['Pclass'], as_index=False).mean().sort_values(by='Survived', ascending=False))
print('*'*50)
print(train[['SibSp','Survived']].groupby(['SibSp'],as_index = False).mean().sort_values(by ='SibSp',ascending = False))
print('*'*50)
print(train[['Parch','Survived']].groupby(['Parch'],as_index = False).mean().sort_values(by='Parch',ascending = False))

   Pclass  Survived
0       1  0.629630
1       2  0.472826
2       3  0.242363
**************************************************
   SibSp  Survived
6      8  0.000000
5      5  0.000000
4      4  0.166667
3      3  0.250000
2      2  0.464286
1      1  0.535885
0      0  0.345395
**************************************************
   Parch  Survived
6      6  0.000000
5      5  0.200000
4      4  0.000000
3      3  0.600000
2      2  0.500000
1      1  0.550847
0      0  0.343658


#### 4. Correlation between 'Sex' and 'Survived'

In [9]:
train[['Sex','Survived']].groupby(['Sex'],as_index = False).mean().sort_values(by = 'Sex', ascending = 'True')

Sex  Survived
0  female  0.742038
1    male  0.188908

### Yes we need to include this feature, since the correlation is high when compared to Pclass,SibSp,Parch.
### From above we can find that if traveler is female, then chance for living is 74.2% and if traveler is male , then chance for living is nearly 19%.